File: python-advanced/src/utils/logger.py
import json
from datetime import datetime
from enum import Enum

class LogLevel(Enum):
    INFO = "INFO"
    ERROR = "ERROR"
    WARN = "WARN"

class Logger:
    def _log(self, level: LogLevel, message: str, **meta):
        log_entry = {
            "timestamp": datetime.now().isoformat(),
            "level": level.value,
            "message": message,
            **meta
        }
        print(json.dumps(log_entry))

    def info(self, message: str, **meta):
        self._log(LogLevel.INFO, message, **meta)

    def error(self, message: str, **meta):
        self._log(LogLevel.ERROR, message, **meta)

    def warn(self, message: str, **meta):
        self._log(LogLevel.WARN, message, **meta)

logger = Logger()

File: dotnet-advanced/Utils/Logger.cs
using System;
using System.Collections.Generic;
using System.Text.Json;

namespace AdvancedUserService.Utils
{
    public enum LogLevel
    {
        INFO,
        ERROR,
        WARN
    }

    public class Logger
    {
        public void Log(LogLevel level, string message, Dictionary<string, object>? meta = null)
        {
            var entry = new Dictionary<string, object>
            {
                ["timestamp"] = DateTime.UtcNow.ToString("o"),
                ["level"] = level.ToString(),
                ["message"] = message
            };

            if (meta != null)
            {
                foreach (var kvp in meta)
                {
                    entry[kvp.Key] = kvp.Value;
                }
            }

            Console.WriteLine(JsonSerializer.Serialize(entry));
        }

        public void Info(string message, Dictionary<string, object>? meta = null)
        {
            Log(LogLevel.INFO, message, meta);
        }

        public void Error(string message, Dictionary<string, object>? meta = null)
        {
            Log(LogLevel.ERROR, message, meta);
        }

        public void Warn(string message, Dictionary<string, object>? meta = null)
        {
            Log(LogLevel.WARN, message, meta);
        }
    }
}
